Library/Homebrew/unpack_strategy.rb

```diff
@@ -187,7 +187,7 @@ def extract_nestedly(to: nil, basename: nil, verbose: false, prioritize_extensio
     end
   end
 
-  sig { returns(T::Array[String]) }
+  sig { returns(T.nilable(T::Array[T.any(String, Formula)])) }
```

Comment:
```suggestion
  sig { returns(T.nilable(T::Array[T.any(Cask::Cask, Formula, String)])) }
```

Ha, we were working on this at the same time. Looking at the various `UnpackStrategy` `#dependencies` methods, `Air` also returns an array of `Cask::Cask` objects.

Reply:
Might even need to be
```suggestion
  sig { returns(T.nilable(T.any(T::Array[Cask::Cask], T::Array[Formula]))) }
```

Reply:
`T.nilable` smells a bit here. Anything returning that should probably be returning `[]`.

Reply:
I originally used `T.any(T::Array[Cask::Cask], T::Array[Formula])` in my local implementation, so that makes sense to me (i.e., currently this is either an array of casks or an array of formulae but not an array of casks or formulae).

-----

I was thinking the same thing about `T.nilable` in these method signatures but I hadn't looked into whether the code that calls these methods distinguishes between `nil` and an array. I can at least say that `brew typecheck` didn't complain when I removed `T.nilable` from these methods (when I tested the idea earlier).